lib/common/api/deprecate.js

```diff
@@ -46,6 +46,26 @@ deprecate.log = (message) => {
   }
 }
 
+// Deprecate an event.
+deprecate.event = (emitter, oldName, newName, fn) => {
+  let warned = false
+  return emitter.on(newName, function (...args) {
+    // There are no listeners for this event
+    if (this.listenerCount(oldName) === 0) { return }
+    // noDeprecation set or if user has already been warned
+    if (warned || process.noDeprecation) { return }
+    warned = true
+    // Event name starts with a -, which means it's an internal event.
+    // For deprecation, that means the event cannot be used anymore.
+    if (!newName.startsWith('-')) { deprecate.warn(`'${oldName}' event`, `'${newName}' event`) } else { deprecate.log(`'${oldName}' event has been deprecated.`) }
```

Comment:
:-( Linter fix made this line long.

Reply:
You mean you cannot use line breaks here?

Reply:
I tried having if and else on separate lines. But the linter auto fixed to this. Maybe I can trick it somehow, but it's probably fine.

Reply:
I thought we have only one linter for all js files.
Let's ignore the linter if it suggests this nonsense ), and write it like that:

```js
if () {
  ...
} else {
  ...
}
```